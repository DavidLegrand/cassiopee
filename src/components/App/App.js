import { useState, useEffect } from 'react'
import "./App.css";
import { Row, Col } from "react-bootstrap";
import Header from "components/Header";
import Main from "components/Main";

function App() {
  const [open, setOpen] = useState(false);
  const [scrolled, setScrolled] = useState(false)

  const handleScroll = () => {
    setScrolled(window.scrollY > 66)
  }

  useEffect(() => {
    setScrolled(window.scrollY > 66)
    window.addEventListener('scroll', handleScroll, true)
    return () => {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [])

  return (
    <>
      <Header scrolled={scrolled} setOpen={setOpen} open={open}></Header>
      <Row noGutters>
        <Col>
          <Main open={open}></Main>
        </Col>
      </Row>
    </>
  );
}

export default App;
