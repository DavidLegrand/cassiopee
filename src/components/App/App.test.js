import App from "./App";
import { shallow } from "enzyme";

describe("App", () => {
  test("Renders", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
