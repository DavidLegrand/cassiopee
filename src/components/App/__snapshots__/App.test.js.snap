// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders 1`] = `ShallowWrapper {}`;
